das/stats.go

```diff
@@ -0,0 +1,59 @@
+package das
+
+import (
+	"sync"
+	"time"
+)
+
+// TODO(@walldiss): discuss new stats
+
+// state collects information about the DASer process. Currently, there are
+// only two sampling routines: the main sampling routine which performs sampling
+// over current network headers, and the `catchUp` routine which performs sampling
+// over past headers from the maxKnown sampled checkpoint.
+type stats struct {
+	sampleLk sync.RWMutex
+	sample   RoutineState // tracks information related to the main sampling routine
+
+	catchUpLk sync.RWMutex
+	catchUp   JobInfo // tracks information related to the `catchUp` routine
+}
+
+// RoutineState contains important information about the state of a
+// current sampling routine.
+type RoutineState struct {
+	// reports if an error has occurred during the routine's
```

Comment:
You could add the struct fields' names to the beginning of the comments so they are all godoc'ed